contracts/Litigate.sol

```diff
@@ -0,0 +1,240 @@
+    /// @param signature EIP-712 signature
+    function litigateContent(
+        ContentToLitigate calldata content,
+        bytes calldata signature
+    ) public returns (bool) {
+        require( block.timestamp > content.timestamp + TIME_THRESHOLD, "Time threshold has not passed yet");
+        require( verifyMetaTxContent(content, signature), "Invalid signature");
+        // check if content is already registered
+        string memory firstUrl = contents.list[0].url;
+        require( keccak256(bytes(content.url)) != keccak256(bytes(firstUrl)), "Initial content is not litigable" );
+        uint256 contentIndex = contents.ids[content.url];
+        if (contentIndex != 0){
+            // content exists, check if constant values are correct
+            Content memory existingContent = contents.list[contentIndex];
+            bool isTitleCorrect = keccak256(bytes(content.title)) == keccak256(bytes(existingContent.title));
+            bool isSubmittedByCorrect = content.submittedBy == existingContent.submittedBy;
+            bool isTagIdsCorrect = content.tagIds.length == existingContent.tagIds.length;
+            bool isTagNameCorrect = false;
+            if (isTagIdsCorrect){
+                for (uint256 i = 0; i < content.tagIds.length; i++) {
+                    isTagNameCorrect =
+                        keccak256(bytes(content.tagIds[i]))
+                        ==
+                        keccak256(bytes(tags.list[existingContent.tagIds[i]].name));
+                    isTagIdsCorrect = isTagIdsCorrect && isTagNameCorrect;
+                }
+            }
+            // if Content values are correct return false because litigation is not correct
+            if (isTitleCorrect && isSubmittedByCorrect && isTagIdsCorrect){
+                return false;
+            }
+        }
+        // content does not exists or values are incorrect, add it or modify it
+        _createContentIfNotExists(
+            content.title,
+            content.url,
+            content.submittedBy,
+            content.likes,
+            content.tagIds
+        );
+        slashBackend(msg.sender);
+        return true;
+    }
+
+    /// @notice Litigate a specific tag (add it and claim slashing)
+    /// @dev For more info about verifying EIP-712
+    /// check https://github.com/aglawson/Meta-Transactions/blob/main/contracts/MinimalForwarder.sol
+    /// @param tag Tag to litigate
+    /// @param signature EIP-712 signature
+    function litigateTag(
+        TagToLitigate calldata tag,
+        bytes calldata signature
+    ) public returns (bool) {
+        require( block.timestamp > tag.timestamp + TIME_THRESHOLD, "Time threshold has not passed yet");
+        require( verifyMetaTxTag(tag, signature), "Invalid signature" );
+        // check if tag is already registered
+        string memory firstTag = tags.list[0].name;
+        require( keccak256(bytes(tag.name)) != keccak256(bytes(firstTag)), "Initial tag is not litigable" );
```

Comment:
lint (multiple instances of extra spaces in this file, not highlighting them all just formatting)